CRM/Contribute/Page/ContributionRecurTab.php

```diff
@@ -0,0 +1,493 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.6                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2015                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2015
+ * $Id$
+ *
+ */
+class CRM_Contribute_Page_ContributionRecurTab extends CRM_Core_Page {
+
+  /**
+   * The action links that we need to display for the browse screen.
+   *
+   * @var array
+   */
+  static $_links = NULL;
+  public $_permission = NULL;
+  public $_contactId = NULL;
+  public $_crid = NULL;
+
+  /**
+   * This method returns the links that are given for recur search row.
+   * currently the links added for each row are:
+   * - View
+   * - Edit
+   * - Cancel
+   *
+   * @param bool $recurID
+   * @param string $context
+   *
+   * @return array
+   */
+  public static function &recurLinks($recurID = FALSE, $context = 'contribution') {
```

Comment:
Hmm - I have been randomly prefixing based on what is already in the code as a quasi-pattern. Would be happy to adopt a non-prefixing going forwards - but that would need to be communicated to all & sundry